### ISSUE TITLE
is expression unimplemented for union types

### ISSUE THREAD

**Issue by thisbefruit** [mel-project/melodeon#18]
These are the relevant definitions:
````
type False = {0}
type True = {1}
type Bool = True | False

type D = {12}
type T = {1}
type DT = D | T
````

Here are the outputs:
````
melorun> let x = 1 in x is Bool
thread 'main' panicked at 'not yet implemented', /home/thisbefruit/.cargo/registry/src/github.com-1ecc6299db9ec823/melodeon-0.4.17/src/codegen.rs:430:30
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

melorun> let x = 1 in x is DT
thread 'main' panicked at 'not yet implemented', /home/thisbefruit/.cargo/registry/src/github.com-1ecc6299db9ec823/melodeon-0.4.17/src/codegen.rs:430:30
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
````

Certain union types in the REPL generate the same problem:
````
melorun> let x = 1 in x is Nat | %[]
thread 'main' panicked at 'not yet implemented', /home/thisbefruit/.cargo/registry/src/github.com-1ecc6299db9ec823/melodeon-0.4.17/src/codegen.rs:430:30
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

melorun> let x = 1 in x is {1} | {2}
thread 'main' panicked at 'not yet implemented', /home/thisbefruit/.cargo/registry/src/github.com-1ecc6299db9ec823/melodeon-0.4.17/src/codegen.rs:430:30
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

melorun> let x = 1 in x is [{1}] | {1}
thread 'main' panicked at 'not yet implemented', /home/thisbefruit/.cargo/registry/src/github.com-1ecc6299db9ec823/melodeon-0.4.17/src/codegen.rs:430:30
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
````

**Comment by nullchinchilla** [comment]
Yes, the current code generation for `is` is highly incomplete. Many different cases need to be filled in

**Comment by nullchinchilla** [comment]
This is fixed in 0.5.2.
